codegen/lib/graphql_swift_gen.rb

```diff
@@ -207,7 +207,10 @@ def deserialize_value_code(field_name, expr, type, untyped: true)
       scalars[type.name].deserialize_expr(expr)
     when 'LIST'
       rethrow = "try " if %w(OBJECT INTERFACE UNION).include?(type.unwrap.kind)
-      "#{rethrow}#{expr}.map { #{deserialize_value_code(field_name, '$0', type.of_type, untyped: !type.of_type.non_null?)} }"
+      untyped = !type.of_type.non_null?
+      sub_statements = "#{rethrow}#{expr}.map { #{deserialize_value_code(field_name, '$0', type.of_type, untyped: untyped)} }"
+      sub_statements += " as [#{swift_json_type(type.of_type)}?]" if untyped
```

Comment:
`swift_json_type` gets the input type to deserialization, but we want the output type.  For example, this would be `[String: Any]` for an OBJECT type, but we want the constructed object for that object type.